src/mjolnir/pbfgraphparser.cc

```diff
@@ -46,6 +46,94 @@ int get_number(const std::string& tag, const std::string& value) { // NOLINT
   return num;
 }
 
+// This class helps to set "culdesac" labels to loop roads correctly.
+class culdesac_processor {
```

Comment:
@maksimandrianov could you add more detailed explanation (as comments) how this class works ?

Reply:
yes, done)

Reply:
thanks